### PR TITLE
Improve the build process

### PR DESCRIPTION
**Pull request by euank**
This improvement brings more speed through not calling go generate every
time, avoids the build-time dependency on godep, mockgen, and goimports,
and otherwise adds a bit of sanity.

I see this as an incremental change. There's still room for further improvement but I think this is a step in the right direction.

r? @samuelkarp

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -70,16 +68,15 @@ misc/certs/ca-certificates.crt:
 	docker build -t "amazon/amazon-ecs-agent-cert-source:make" misc/certs/
 	docker run "amazon/amazon-ecs-agent-cert-source:make" cat /etc/ssl/certs/ca-certificates.crt > misc/certs/ca-certificates.crt
 
-short-test: gogenerate
-	cd agent && godep go test -short -timeout=25s -v -cover ./...
-	@git checkout -- agent/version/version.go
+short-test:
+	. ./scripts/shared_env && go test -short -timeout=25s -v ./agent/...
```
Comment (samuelkarp) [inline Makefile]: Can you change all instances of `. ./scripts/shared_env` to `source ./scripts/shared_env`?

File: scripts/shared_env [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# permissions and limitations under the License.
+
+# This script sets up the GOPATH and PATH that the agent expects, as well as
+# setting any other common environment variables
+# It is expected to be 'sourced' while the CWD is the root of the repository
+
+# Manually setup godep path; avoid build-time dependency on the godep binary
+# The existing GOPATH needs to have the agent itself in it, but shouldn't need
+# anything else
+export GOPATH="`pwd`/agent/Godeps/_workspace/:$GOPATH"
```
Comment (samuelkarp) [inline scripts/shared_env]: Can you change this to `export GOPATH="$(pwd)/agent/Godeps/_workspace/:${GOPATH}"`?  Similar changes for the `PATH` variable below.

**Comment by samuelkarp** [comment]
:+1:, just a few minor nitpicks.

File: Makefile (hunk 1 above)
Comment (euank) [inline Makefile]: I can do it in the bash scripts. Makefile's commands are interpreted with `sh` which does not have the `source` builtin. I don't know of a more readable alternative to `.` in `sh`

Comment (samuelkarp) [inline Makefile]: That's fine then.  There are ways to control which shell make uses, but `.` is probably more readable than changing the shell.  Please change it in the bash scripts.

**Comment by euank** [comment]
Changed per your comments.

**Comment by samuelkarp** [comment]
:+1:

**Pull request merged**